src/Explain.h

```diff
@@ -32,199 +36,206 @@
 
 namespace souffle {
 
-class Explain {
+class ExplainConfig {
 public:
-    Explain(ExplainProvenance& p, bool ncurses, bool silentPrompt, int d = 4)
-            : prov(p), ncurses(ncurses), silentPrompt(silentPrompt), depthLimit(d), output(nullptr),
-              json(false) {}
-    ~Explain() {
-        delete output;
+    /* Deleted copy constructor */
+    ExplainConfig(const ExplainConfig&) = delete;
+
+    /* Deleted assignment operator */
+    ExplainConfig& operator=(const ExplainConfig&) = delete;
+
+    /* Obtain the global ExplainConfiguration */
+    static ExplainConfig& getExplainConfig() {
+        static ExplainConfig _config;
+        return _config;
     }
 
-    void explain() {
-        if (ncurses && !output) {
-            initialiseWindow();
-            std::signal(SIGWINCH, nullptr);
-        }
+    /* Configuration variables */
+    std::ostream* outputStream = nullptr;
+    bool json = false;
+    int depthLimit = 4;
 
-        printPrompt("Explain is invoked.\n");
+private:
+    ExplainConfig() {}
+};
 
-        while (true) {
-            clearDisplay();
-            printPrompt("Enter command > ");
-            std::string line = getInput();
+class Explain {
+public:
+    ExplainProvenance& prov;
 
-            std::vector<std::string> command = split(line, ' ', 1);
+    Explain(ExplainProvenance& p) : prov(p) {}
+    ~Explain() {
+        // close file streama
+        if (ExplainConfig::getExplainConfig().outputStream != nullptr) {
```

Comment:
If Explain can delete things, then they belong to Explain - suggesting that either ExplainConfig should delete the pointer or the pointer should be part of Explain.

It's also safe to delete a nullptr, so the if is not needed.